compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1542,7 +1549,7 @@ public boolean validateKeySpecifier(List<String> fieldNameList, BType constraint
                 return true;
             }
 
-            if (!Symbols.isFlagOn(field.symbol.flags, Flags.REQUIRED)) {
+            if (Symbols.isFlagOn(field.symbol.flags, Flags.OPTIONAL)) {
```

Comment:
Do we have a test like the following sample, to make sure an error is logged when the field is optional?

```ballerina
type Employee record {|
    readonly int id?;
    readonly string name;
    string department;
|};

public function main() {
    table<Employee> key(id) tb; // error
}
```

Reply:
Yes.
https://github.com/ballerina-platform/ballerina-lang/blob/2c6357b0f04b7e852bb9c9f0813084d8706ba5f2/tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal#L75